opentreemap/opentreemap/celery.py

```diff
@@ -16,3 +19,12 @@
 # pickle the object when using Windows.
 app.config_from_object('django.conf:settings')
 app.autodiscover_tasks(lambda: settings.INSTALLED_APPS)
+
+
+@task_failure.connect
+def handle_task_failure(**kw):
+    rollbar_settings = getattr(settings, 'ROLLBAR', {})
+    access_token = rollbar_settings.get('access_token')
+    if access_token:
+        rollbar.init(access_token, rollbar_settings['environment'])
```

Comment:
Thanks for bringing this up, I was confused about this.

Initially when I read https://github.com/rollbar/pyrollbar/blob/master/rollbar/contrib/django/middleware.py#L92 I thought that rollbar was being initialized once per request and so I copied that pattern. I see now in https://docs.djangoproject.com/en/1.9/topics/http/middleware/#init that this is actually happening once per process/thread, so it would be good to follow suit.

I do not know if celery will load the middleware and initialize rollbar. Even if it does, it creates an implicit dependency. I think it's probably better to initialize it once in `celery.py`, and once through the middleware. It looks like rollbar keeps track of inits with internal state and doesn't expose a way to check (https://github.com/rollbar/pyrollbar/blob/master/rollbar/__init__.py#L300). This exposes us to the `re-init` warning, but is probably OK, it's better than doing it on every exception. Does that make sense?

I do not understand what you mean by "bother keeping our own flag".